src/tests/GfxBatchHbaoTest.cpp

```diff
@@ -0,0 +1,352 @@
+     esp::gfx_batch::HbaoConfiguration{}.intensity(),
+     esp::gfx_batch::HbaoConfiguration{}.radius(),
+     esp::gfx_batch::HbaoConfiguration{}.blurSharpness(), 2.0f, 2.0f},
+    {"classic, orthographic, strong effect",
+     "hbao_tests/van-gogh-room.hbao-classic-strong-ortho.png", true, false,
+     esp::gfx_batch::HbaoConfiguration{}.flags(), 2.0f, 1.5f, 10.0f, 0.0f,
+     0.0f},
+
+    {"cache-aware, orthographic, defaults",
+     "hbao_tests/van-gogh-room.hbao-cache-ortho.png", false, false,
+     esp::gfx_batch::HbaoConfiguration{}.flags(),
+     esp::gfx_batch::HbaoConfiguration{}.intensity(),
+     esp::gfx_batch::HbaoConfiguration{}.radius(),
+     esp::gfx_batch::HbaoConfiguration{}.blurSharpness(), 0.0f, 0.0f},
+    {"cache-aware, orthographic, AO special blur",
+     "hbao_tests/van-gogh-room.hbao-cache-sblur-ortho.png", false, false,
+     esp::gfx_batch::HbaoConfiguration{}.flags() |
+         esp::gfx_batch::HbaoFlag::UseAoSpecialBlur,
+     esp::gfx_batch::HbaoConfiguration{}.intensity(),
+     esp::gfx_batch::HbaoConfiguration{}.radius(),
+     esp::gfx_batch::HbaoConfiguration{}.blurSharpness(), 2.0f, 2.0f},
+    {"cache-aware, orthographic, strong effect",
+     "hbao_tests/van-gogh-room.hbao-cache-strong-ortho.png", false, false,
+     esp::gfx_batch::HbaoConfiguration{}.flags(), 2.0f, 1.5f, 10.0f, 0.0f,
+     0.0f},
+    {"cache-aware, orthographic, layered with image load store",
+     "hbao_tests/van-gogh-room.hbao-cache-layered-ortho.png", false, false,
+     esp::gfx_batch::HbaoConfiguration{}.flags() |
+         esp::gfx_batch::HbaoFlag::LayeredImageLoadStore,
+     esp::gfx_batch::HbaoConfiguration{}.intensity(),
+     esp::gfx_batch::HbaoConfiguration{}.radius(),
+     esp::gfx_batch::HbaoConfiguration{}.blurSharpness(), 4.5f, 0.11f},
+    {"cache-aware, orthographic, layered with geometry shader",
+     "hbao_tests/van-gogh-room.hbao-cache-geom-ortho.png", false, false,
+     esp::gfx_batch::HbaoConfiguration{}.flags() |
+         esp::gfx_batch::HbaoFlag::LayeredGeometryShader,
+     esp::gfx_batch::HbaoConfiguration{}.intensity(),
+     esp::gfx_batch::HbaoConfiguration{}.radius(),
+     esp::gfx_batch::HbaoConfiguration{}.blurSharpness(), 4.5f, 0.11f},
+};
+
+GfxBatchHbaoTest::GfxBatchHbaoTest() {
+  addTests({&GfxBatchHbaoTest::generateTestData});
+
+  addInstancedTests({&GfxBatchHbaoTest::test},
+                    Cr::Containers::arraySize(TestData));
+}
+
+constexpr Mn::Vector2i Size{256, 256};
+void GfxBatchHbaoTest::generateTestData() {
+  Cr::PluginManager::Manager<Mn::Trade::AbstractImporter> importerManager;
+  Cr::Containers::Pointer<Mn::Trade::AbstractImporter> importer =
+      importerManager.loadAndInstantiate("AnySceneImporter");
+  CORRADE_VERIFY(importer);
+
+  /* magnum-sceneconverter van-gogh-room.glb --concatenate-meshes
+   * van-gogh-room.mesh.ply */
+  CORRADE_VERIFY(importer->openFile(Cr::Utility::Path::join(
+      TEST_ASSETS, "hbao_tests/van-gogh-room.mesh.ply")));
```

Comment:
eh, 500 kb isn't that bad, is it? :)

Reply:
It's more about the principle / general workflow. The original file wasn't commited, so I didn't put this one here either. And (again), the file isn't needed for verifying the HBAO workflow, it's only needed once, to generate the input data locally. In other words, the test-data-generation case could even be a completely separate executable / python script that isn't being run on the CI at all, but this way it felt to me that it's less prone to code rot and getting out of sync with the actual images.

With the "isn't that bad" mindset the `van-gogh-room.glb` could be in the repo already instead of being downloaded separately, allowing the single-joined-mesh variant of to be generated by the test on-the-fly. Which would be the cleanest way. Commiting just the derived file and not the original doesn't make a lot of sense to me.

Reply:
Ply has been removed.